ConnectorRhino/ConnectorRhino/ConnectorRhinoShared/UI/MappingBindingsRhino.cs

```diff
@@ -26,41 +26,44 @@ public MappingBindingsRhino()
 
   public override MappingSelectionInfo GetSelectionInfo()
   {
-    try
+    List<RhinoObject> selection = RhinoDoc.ActiveDoc.Objects.GetSelectedObjects(false, false)?.ToList();
+    if (selection is null)
     {
-      var selection = RhinoDoc.ActiveDoc.Objects.GetSelectedObjects(false, false).ToList();
-      var result = new List<Schema>();
+      return new MappingSelectionInfo(new List<Schema>(), 0);
+    }
 
-      foreach (var obj in selection)
-      {
-        var schemas = GetObjectSchemas(obj);
+    var result = new List<Schema>();
+    foreach (var obj in selection)
+    {
+      List<Schema> schemas = GetObjectSchemas(obj);
 
-        if (!result.Any())
-        {
-          result = schemas;
-        }
-        else
+      if (!result.Any())
+      {
+        result = schemas;
+      }
+      else
+      {
+        //intersect lists
+        //TODO: if some elements already have a schema and values are different
+        //we should default to an empty schema, instead of potentially restoring the one with values
+        try
         {
-          //intersect lists
-          //TODO: if some elements already have a schema and values are different
-          //we should default to an empty schema, instead of potentially restoring the one with values
           result = result.Where(x => schemas.Any(y => y.Name == x.Name)).ToList();
         }
-
-        //incompatible selection
-        if (!result.Any())
-        {
-          return new MappingSelectionInfo(new List<Schema>(), selection.Count);
-        }
+        catch (ArgumentNullException) { }
       }
 
-      return new MappingSelectionInfo(result, selection.Count);
-    }
-    catch (Exception ex)
-    {
-      SpeckleLog.Logger.Error(ex, "Could not get selection info: {exceptionMessage}", ex.Message);
-      return new MappingSelectionInfo(new List<Schema>(), 0);
+      //incompatible selection
+      if (!result.Any())
+      {
+        return new MappingSelectionInfo(new List<Schema>(), selection.Count);
+      }
     }
+
+    return new MappingSelectionInfo(result, selection.Count);
+
+    // todo: previously entire block was try catched: SpeckleLog.Logger.Error(ex, "Could not get selection info: {exceptionMessage}", ex.Message);
```

Comment:
I'm not a fan of adding more todos to the code, I would prefer we raise tickets and discus getting them onto the sprint if we have actions to follow up

Reply:
Due to complications with Avalonia and DUI2, I'm not entirely comfortable completely removing entire blocks wrapped in a trycatch - I can rethrow with `IsFatal()` as previously suggested however since this not a topmost exception